apps/web/src/views/Notifications/containers/OnBoardingView.tsx

```diff
@@ -51,35 +53,52 @@ function OnboardingButton({ onClick, loading, isOnBoarded, account, isReady }: I
   )
 }
 
-const OnBoardingView = ({ identityKey, handleRegistration, isReady }: IOnBoardingProps) => {
+const OnBoardingView = ({
+  identityKey,
+  handleRegistration,
+  isReady,
+  isSubscribing,
+  setIsSubscribing,
+}: IOnBoardingProps) => {
+  const [isOboarding, setIsOnBoarding] = useState<boolean>(false)
   const toast = useToast()
   const { t } = useTranslation()
   const { address: account } = useAccount()
-  const { subscribe, isSubscribing } = useManageSubscription(`eip155:1:${account}`)
-  const { sendPushNotification, requestNotificationPermission, subscribeToPushNotifications } =
-    useSendPushNotification()
+  const { subscribe } = useManageSubscription(`eip155:1:${account}`)
+  const { sendPushNotification, subscribeToPushNotifications } = useSendPushNotification()
 
   const handleSubscribe = useCallback(async () => {
+    setIsSubscribing(true)
     try {
-      await subscribe()
       await subscribeToPushNotifications()
+      await subscribe()
       setTimeout(async () => {
         await sendPushNotification(BuilderNames.OnBoardNotification, [], `eip155:1:${account}`)
-        await requestNotificationPermission()
-      }, 1500)
+      }, 1200)
+    } catch (error) {
+      const errMessage = parseErrorMessage(Events.SubscriptionRequestError, error)
+      toast.toastError(Events.SubscriptionRequestError.title, errMessage)
+    }
+  }, [account, toast, sendPushNotification, subscribe, subscribeToPushNotifications, setIsSubscribing])
+
+  const handleOnBoarding = useCallback(async () => {
+    setIsOnBoarding(true)
+    try {
+      await handleRegistration()
     } catch (error) {
       const errMessage = parseErrorMessage(Events.SubscriptionRequestError, error)
       toast.toastError(Events.SubscriptionRequestError.title, errMessage)
     }
-  }, [account, toast, sendPushNotification, subscribe, subscribeToPushNotifications, requestNotificationPermission])
+    setIsOnBoarding(false)
+  }, [setIsOnBoarding, handleRegistration, toast])
 
   const handleAction = useCallback(
-    (e: React.MouseEvent<HTMLDivElement | HTMLButtonElement>) => {
+    async (e: React.MouseEvent<HTMLDivElement | HTMLButtonElement>) => {
       e.stopPropagation()
-      if (!identityKey) handleRegistration()
-      else handleSubscribe()
+      if (!identityKey) await handleOnBoarding()
```

Comment:
What's the reason for adding await here?

Reply:
none removed